exercise3.js

```diff
@@ -86,50 +86,81 @@
-
-
-console.log(mapObj.has({company : "TEKsystems"}));  
+mapObj.set(objAsKey = {company : "TEKsystems"},"object"); // this line has now been refactored
+mapObj.set(2020, 'number');
+mapObj.set('hello', 'string');
+mapObj.set([1,2,3], 'array');
+mapObj.set(false, 'boolean');
+mapObj.set(undefined, 'undefined');
+
+console.log(mapObj.has({company : "TEKsystems"}));
 //The above console.log() statmeent returns false.  Write another console.log() statement explaining why this line of code prints false.  Refactor the code on line 106, so you can successfully check to see if {company : "TEKsystems"} exists in the mapObj.
 
-//your code...
+console.log('The above console.log() in this problem is returning false because when we set an object as a key we need to provide a reference. If we don\'t we are setting one object as a key and will fail to successfuly reference that key when using the has method in Map because our argument for the has method is a completely new object even though it may seem as being the same since the values for both objects are the same i.e. {company : "TEKsystems"} === {company : "TEKsystems"} results to false.');
+
 
 //loop through the mapObj and create a new array of only the data types, leaving out the example keys of the mapObj.  Use array methods to do this.  Example output : ['string',number','boolean',array','object']
+const newMapObj = [];
+mapObj.forEach((el) => {newMapObj.push(el)});
+console.log(newMapObj);
 
 
 /************************************************************* */
 //Problem 11:
 
 let ones = [1,11,111,1111,11111,111111,1111111,11111111,111111111,1111111111];
 //reverse the array, without modifying the ones array.
+const onesReversed = ones.slice().reverse();
+console.log(ones);
+console.log(onesReversed);
 
 
 /************************************************************* */
 //Problem 12:
 //create a function called performer(cb) that takes in a callback function and runs that callback function.  It should return the output of the callback function.
 
 function performer(cb) {
-    //code goes here
+    return cb();
 }
 
 
 /************************************************************* */
 //Bonus assignment:
 //research a new feature of ES6+ and create an example of it's use case here.  Be sure to write comments explaining what the feature is and why it is useful.
+const myName = 'Jose Moreno'
+const injectHTMLGreeting = `<div id="greeting">
+                                <h1>Hello my name is ${myName}</h1>
+                            </div>`
+document.querySelector('body').insertAdjacentHTML("afterbegin", injectHTMLGreeting);
+
+// I learned this week of Multi-line Strings using template literals (backticks). I created a snippet of code above to best describe one reason why this feature is useful. 
+// As you may see, the variable injectHTMLGreeting contains our multi-line string which in this multi-line format we clearly can see which HTML tags we are going to be adding to our html page. 
```

Comment:
Overall great work on this assignment. One thing to do in the future though is to make sure all your solutions are logging the result of your work so we can run it and see your successes.